docs/proposals/ThumbnailGallery/README.md

```diff
@@ -0,0 +1,60 @@
+# Component Name
+
+The `ThumbnailGallery` component will accept a list of attachments for images
+and display them as thumbnails. This would also have an option to show the
+images in a LightBox component on click of those images.
+
+## Accessibility
+
+Assistive technology should be able to see the images and know the image name
+based on the `name` property of each attachment.
+
+## Responsiveness & Mobile
+
+This component show maximum number of images possible in a line and show the
+count of images left to show.
+
+## Wireframe
+
+[mockup](mockup.png)
+
+## Interface
+
+```
+<ThumbnailGallery
+    attachments={[{
```

Comment:
Perhaps `images` since these should always be images?

Reply:
Oh that's something I am not sure, assuming the case where we want to just show just an icon for pdf and video documents. But the light box would just show the images.

Reply:
> Just confirmed with @chriscarline that, for the use case in activity feed, we do need to show the icons for pdfs and videos in thumbnail view. But LightBox would exclude non image attachments.

I think this might lead to confusing behaviour (the sequence of images in the lightbox wouldn't match the thumbnails, and that would be worse if not all of the thumbnails are shown).

@chriscarline & @chris-at-jobber - would some Figma work help here? It feels like the full requirements of the component aren't completely captured in the text... (see other discussion in this PR for examples.)

Reply:
Discussed about this with @chriscarline. We feel the ideal approach is that we would want to see a pdf preview in LightBox as well but since that is not currently supported by LightBox (I am not really sure, does it?) we are okay to not display pdfs in LightBox and maybe onclick of a pdf will open them in a new tab just like how it behaves in notes section.